### PR TITLE
move tsfile-format.properties to conf directory

### PR DESCRIPTION
**Pull request by MyXOF**

### CODE REVIEW

**Comment by MyXOF** [comment]
solve #71

**Comment by jixuan1989** [comment]
The code is fine. 
Update the get startted document to tell people how to set the url if they want.

File: src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileConfig.java [hunk 1]
```diff
@@ -129,7 +129,7 @@
 
     public static final String CONFIG_NAME = "tsfile-format.properties";
 
-    public static String CONFIG_DEFAULT_PATH = "src/test/resources/" + CONFIG_NAME;
+//    public static String HOME_DEFAULT_PATH = "conf/" + CONFIG_NAME;
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileConfig.java]: delete it

File: src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java [hunk 2]
```diff
@@ -1,88 +1,124 @@
...
+	private void multiplicityWarning(String resource, ClassLoader classLoader) {
+		try {
+			Set<URL> urlSet = Loader.getResources(resource, classLoader);
+			if (urlSet != null && urlSet.size() > 1) {
+				LOGGER.warn("Resource [{}] occurs multiple times on the classpath", resource);
+				for (URL url : urlSet) {
+					LOGGER.warn("Resource [{}] occurs at [{}]", resource, url.toString());
+				}
+			}
+		} catch (IOException e) {
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java]: How to resolve exception?

File: src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java [hunk 3]
```diff
@@ -1,88 +1,124 @@
...
+			conf.compressor = properties.getProperty("compressor", conf.compressor);
+		} catch (IOException e) {
+			LOGGER.warn("Cannot load config file because {}, use default configuration", e.getMessage());
+		} catch (Exception e) {
+			LOGGER.error("Loading settings {} failed because {}", url, e.getMessage());
+		} finally {
+			if (inputStream != null) {
+				try {
+					inputStream.close();
+					inputStream = null;
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java]: inputStream is never used, no need to set null

File: src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java [hunk 4]
```diff
@@ -1,88 +1,124 @@
...
+				} catch (IOException e) {
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java]: Catch the exception and then what?

File: src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java [hunk 5]
```diff
@@ -1,88 +1,124 @@
...
+			conf.floatPrecision = Integer.parseInt(properties.getProperty("float_precision", conf.floatPrecision + ""));
+			conf.timeSeriesEncoder = properties.getProperty("time_series_encoder", conf.timeSeriesEncoder);
+			conf.valueEncoder = properties.getProperty("value_encoder", conf.valueEncoder);
+			conf.compressor = properties.getProperty("compressor", conf.compressor);
+		} catch (IOException e) {
+			LOGGER.warn("Cannot load config file because {}, use default configuration", e.getMessage());
+		} catch (Exception e) {
+			LOGGER.error("Loading settings {} failed because {}", url, e.getMessage());
+		} finally {
+			if (inputStream != null) {
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java]: inputStream is never null, this if sentence is not needed

File: src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java [hunk 6]
```diff
@@ -1,88 +1,124 @@
...
-            if (inputStream != null) {
-                try {
-                    inputStream.close();
-                    inputStream = null;
-                } catch (IOException e) {
-                }
-            }
-        }
-    }
+	public static void main(String[] args) {
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java]: remove this main method

File: src/main/java/cn/edu/tsinghua/tsfile/timeseries/utils/Loader.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+	}
+	
+	public static ClassLoader getClassLoaderOfObject(Object o){
+		if(o == null){
+			throw new NullPointerException("Input object cannot be null");
+		}
+		return getClassLoaderOfClass(o.getClass());
+	}
+	
+	public static ClassLoader getClassLoaderOfClass(final Class<?> clazz){
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/tsfile/timeseries/utils/Loader.java]: this method should be private

File: src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java (hunk 2 above)
Comment (kr11) [reply src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java]: may it select the first url?

File: src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java [hunk 8]
```diff
@@ -1,88 +1,124 @@
...
+		}
 
-    public TSFileConfig getConfig() {
-        return conf;
-    }
+		LOGGER.info("Start to read config file {}", url);
+		Properties properties = new Properties();
+		try {
+			properties.load(inputStream);
+			conf.groupSizeInByte = Integer
```
Comment (kr11) [inline src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java]: Here we explicitly load some parameters like .etc, It's not scalable if new parameters appear. why we don't use the YAML importer or other public utils?

**Comment by kr11** [comment]
+1

**Pull request merged**